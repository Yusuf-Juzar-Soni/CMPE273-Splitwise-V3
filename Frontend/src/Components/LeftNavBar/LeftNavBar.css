.flex-column{
background-color:rgb(246, 252, 246) !important;
height: 100vh;
width: 200px;     /* Set the width of the sidebar */
z-index: 1;      /* Stay on top of everything */
     /* Stay at the top */
padding-top: 10px;

}
