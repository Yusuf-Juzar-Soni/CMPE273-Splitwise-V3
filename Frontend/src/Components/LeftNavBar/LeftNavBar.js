import { React, useEffect, useState } from "react";
import Nav from "react-bootstrap/Nav";
import axios from "axios";
import { useHistory } from "react-router-dom";
import { useSelector } from "react-redux";
const queryString = require("query-string");
import { useLocation } from "react-router-dom";
import "./LeftNavBar.css";
import {
  Button,
  Grid,
  Row,
  Col,
  ListGroup,
  Form,
  Card,
  Modal,
} from "react-bootstrap";
import backendServer from "../../webConfig";

function LeftNavBar() {
  const history = useHistory();
  const emailID = useSelector((state) => state.isLogged.email);

  const [group_names, group_namesChange] = useState([]);
  const [change, setHandleChange] = useState(false);
  const [list, setListDisplay] = useState([]);
  const location = useLocation();
  const parsed = queryString.parse(location.search);
  function retriveGroups(user_email1) {
    const user_email = { email: user_email1 };
    axios
      .post(`${backendServer}/dashboard`, user_email)
      .then((response) => {
        console.log(response.data);
        group_namesChange(response.data);
        console.log(group_names);
      })
      .catch((err) => {
        if (err.response && err.response.data) {
          alert(err.response.data.message);
        }
      });
  }

  function redirectToGroup(redirectGroup, emailId) {
    history.push({
      pathname: "/groupsdisplay",
      search: `?groupname=` + redirectGroup + `&email=` + emailId,
    });
  }

  function redirectToDashboard(emailId) {
    history.push({
      pathname: "/dash",
      search: `?email=${emailId}`,
    });
  }

  function redirectToCreate(emailId) {
    history.push({
      pathname: "/creategroup",
      search: `?email=${emailId}`,
    });
  }

  function redirectToActivity(emailId) {
    history.push({
      pathname: "/activitydisplay",
      search: `?email=${emailId}`,
    });
  }

  function redirectToInvites(emailId) {
    history.push({
      pathname: "/invitedisplay",
      search: `?email=${emailId}`,
    });
  }
  const handleSearch = (e, groups) => {
    setHandleChange(true);
    console.log(groups);
    let final_list = groups.filter((names) => {
      return names.includes(e.target.value);
    });
    setListDisplay(final_list);
  };

  useEffect(() => {
    retriveGroups(parsed.email);
  }, [location]);

  return (
    <div>
      <div>
        <Nav defaultActiveKey="/home" className="flex-column" color="red">
          <Nav.Link onClick={(event) => redirectToDashboard(parsed.email)}>
            Dashboard
          </Nav.Link>
          <Nav.Link
            data-testid="Activity"
            onClick={(event) => redirectToActivity(parsed.email)}
          >
            Activity
          </Nav.Link>
          <Nav.Link>Groups</Nav.Link>
          <input
            type="text"
            onChange={(e) => handleSearch(e, group_names)}
          ></input>
          {change == true && (
            <ListGroup className="list-group-design">
              {list.map((item) => (
                <Button
                  value={item}
                  key={item}
                  variant="link"
                  onClick={(event) =>
                    redirectToGroup(event.currentTarget.value, parsed.email)
                  }
                >
                  {item}
                </Button>
              ))}
            </ListGroup>
          )}

          {change == false && (
            <ListGroup className="list-group-design">
              {group_names.map((item) => (
                <Button
                  value={item}
                  key={item}
                  variant="link"
                  onClick={(event) =>
                    redirectToGroup(event.currentTarget.value, parsed.email)
                  }
                >
                  {item}
                </Button>
              ))}
            </ListGroup>
          )}
          <Nav.Link onClick={(event) => redirectToCreate(parsed.email)}>
            Create Group
          </Nav.Link>
          <Nav.Link onClick={(event) => redirectToInvites(parsed.email)}>
            Invite List
          </Nav.Link>
        </Nav>
      </div>
    </div>
  );
}
export default LeftNavBar;
