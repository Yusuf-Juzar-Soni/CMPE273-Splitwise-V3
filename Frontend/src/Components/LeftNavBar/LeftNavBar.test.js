import React from "react";
import { render, cleanup } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
// import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import "@testing-library/jest-dom";

import combReducers from "../../reducers";
import LeftNavBar from "./LeftNavBar";

const store = createStore(combReducers);

const component = TestRenderer.create(
  <Provider store={store}>
    <MemoryRouter>
      <LeftNavBar></LeftNavBar>
    </MemoryRouter>
  </Provider>
);

afterEach(cleanup);

it("renders", async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test("Check for header", async () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter>
        <LeftNavBar></LeftNavBar>
      </MemoryRouter>
    </Provider>
  );
  expect(getByTestId("Activity")).toHaveTextContent("Activity");
});
