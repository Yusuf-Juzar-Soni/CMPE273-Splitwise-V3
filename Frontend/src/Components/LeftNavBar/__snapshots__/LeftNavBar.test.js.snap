// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div>
    <div
      className="flex-column nav"
      color="red"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Dashboard
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        data-testid="Activity"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Activity
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Groups
      </a>
      <input
        onChange={[Function]}
        type="text"
      />
      <div
        className="list-group-design list-group"
        onKeyDown={[Function]}
      />
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Create Group
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Invite List
      </a>
    </div>
  </div>
</div>
`;
